packages/@expo/cli/src/api/rest/client.ts

```diff
@@ -47,6 +47,25 @@ export class UnexpectedServerError extends Error {
   readonly name = 'UnexpectedServerError';
 }
 
+/**
+ * An error defining that the server didn't return the expected error JSON information.
+ * The only 'expected' place for this is in testing, all other cases are bugs with the client.
+ */
+export class UnexpectedServerData extends Error {
+  readonly name = 'UnexpectedServerData';
+}
+
+/** Validate the response json contains `.data` property, or throw an unexpected server data error */
+export function getResponseDataOrThrow<T = any>(json: unknown): T {
+  if (!!json && typeof json === 'object' && 'data' in json) {
+    return json.data as T;
+  }
+
+  throw new UnexpectedServerData(
+    !!json && typeof json === 'object' ? JSON.stringify(json) : 'Unknown data received from server.'
+  );
+}
+
```

Comment:
This is because `response.json` now returns `Promise<unknown>`. It's a very simple check to see if `.data` is present, and returns the expected data property as typed value. See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36868#issuecomment-512989650